plotly/tests/test_core/test_plotly/test_plot.py

```diff
@@ -66,3 +70,187 @@ def test_plot_empty_data(self):
         py.sign_in('PlotlyImageTest', '786r5mecv0')
         self.assertRaises(PlotlyEmptyDataError, py.plot, [],
                           filename='plot_invalid')
+
+    def test_plot_sharing_argument(self):
+
+        # Raise an error if sharing argument is incorrect
+        # correct arguments {'public, 'private', 'secret'}
+
+        fig = {
+            'data': [
+                {
+                    'x': [1, 2, 3],
+                    'y': [2, 1, 2]
+                }
+            ]
+        }
```

Comment:
:cow2: mostly personal preference, but i we usually favor tighter formatting for this:

```
fig = {'data': [{'x': [1, 2, 3], 'y': [2, 1, 2]}]}
```

Fits on a single line i think and would probably improve readability.